.github/workflows/new_release.yml

```diff
@@ -1,36 +1,53 @@
-# This is a basic workflow to help you get started with Actions
-
 name: CI new release
 
-# Controls when the workflow will run
 on:
-  # Triggers the workflow on push or pull request events but only for the "master" branch
-  push:
-    branches: [ "master" ]
-  pull_request:
-    branches: [ "master" ]
-
-  # Allows you to run this workflow manually from the Actions tab
   workflow_dispatch:
+    inputs:
+      version:
+        description: Released new version of Realm (0.X.0+beta).
+        required: true
+        type: string
 
-# A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
-  # This workflow contains a single job called "build"
-  build:
-    # The type of runner that the job will run on
+   prepare-pr:
     runs-on: ubuntu-latest
 
-    # Steps represent a sequence of tasks that will be executed as part of the job
     steps:
-      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
       - uses: actions/checkout@v3
 
-      # Runs a single command using the runners shell
-      - name: Run a one-line script
-        run: echo Hello, world!
+      - name: Update realm_dart in pubspec.yaml
+        uses: jacobtomlinson/gha-find-replace@v2
```

Comment:
Why are we doing this manually.
A simple `dart pub upgrade --major-versions` will do the correct things for us